ts/src/bingx.ts

```diff
@@ -0,0 +1,2342 @@
+import { sha256 } from './static_dependencies/noble-hashes/sha256.js';
+import { DECIMAL_PLACES } from './base/functions/number.js';
+import { Int, OrderSide } from './base/types.js';
+
+//  ---------------------------------------------------------------------------
+
+export default class bingx extends Exchange {
+    describe () {
+        return this.deepExtend (super.describe (), {
+            'id': 'bingx',
+            'name': 'BingX',
+            'countries': [ 'US' ], // North America, Canada, the EU, Hong Kong and Taiwan
+            // cheapest is 60 requests a minute = 1 requests per second on average => ( 1000ms / 1) = 1000 ms between requests on average
+            'rateLimit': 1000,
+            'version': 'v1',
+            'certified': true,
+            'pro': true,
+            'has': {
+                'CORS': undefined,
+                'spot': true,
+                'margin': true,
+                'swap': true,
+                'future': false,
+                'option': false,
+                'cancelAllOrders': true,
+                'cancelOrder': true,
+                'cancelOrders': true,
+                'createOrder': true,
+                'fetchBalance': true,
+                'fetchClosedOrders': true,
+                'fetchDeposits': true,
+                'fetchFundingRate': true,
+                'fetchFundingRateHistory': true,
+                'fetchLeverage': true,
+                'fetchMarkets': true,
+                'fetchOHLCV': true,
+                'fetchOpenInterest': true,
+                'fetchOpenOrders': true,
+                'fetchOrder': true,
+                'fetchOrderBook': true,
+                'fetchPositions': true,
+                'fetchTicker': true,
+                'fetchTickers': true,
+                'fetchTime': true,
+                'fetchTrades': true,
+                'fetchTransfers': true,
+                'fetchWithdrawals': true,
+                'setLeverage': true,
+                'setMagin': true,
+                'setMarginMode': true,
+                'transfer': true,
+            },
+            'hostname': 'bingx.com',
+            'urls': {
+                'logo': '',
+                'api': {
+                    'spot': 'https://open-api.{hostname}/openApi',
+                    'swap': 'https://open-api.{hostname}/openApi',
+                    'contract': 'https://open-api.{hostname}/openApi',
+                },
+                'www': 'https://bingx.com/',
+                'doc': 'https://bingx-api.github.io/docs/',
+                'referral': {
+                },
+                'fees': {
+                    'trading': {
+                        'tierBased': true,
+                    },
+                },
+            },
+            'requiredCredentials': {
+                'apiKey': true,
+                'secret': true,
+            },
+            'api': {
```

Comment:
we're missing this endpoint I think we might be able to implement `fetchMyTrades` with it 
![image](https://github.com/ccxt/ccxt/assets/43336371/7ec21818-1c7f-4a80-879c-90818a3ce6b5)